src/containers/NonBitGoRecoveryCoin/ArbethTokenForm.tsx

```diff
@@ -0,0 +1,167 @@
+import { Form, FormikHelpers, FormikProvider, useFormik } from 'formik';
+import { Link } from 'react-router-dom';
+import * as Yup from 'yup';
+import {
+  Button,
+  FormikPasswordfield,
+  FormikSelectfield,
+  FormikTextarea,
+  FormikTextfield,
+} from '~/components';
+
+const validationSchema = Yup.object({
+  backupKey: Yup.string().required(),
+  gasLimit: Yup.number()
+    .typeError('Gas limit must be a number')
+    .integer()
+    .positive('Gas limit must be a positive integer')
+    .required(),
+  gasPrice: Yup.number()
+    .typeError('Gas price must be a number')
+    .integer()
+    .positive('Gas price must be a positive integer')
+    .required(),
+  krsProvider: Yup.string()
+    .oneOf(['keyternal', 'bitgoKRSv2', 'dai'])
+    .label('Key Recovery Service'),
+  recoveryDestination: Yup.string().required(),
+  tokenContractAddress: Yup.string().required(),
+  userKey: Yup.string().required(),
+  walletContractAddress: Yup.string().required(),
+  walletPassphrase: Yup.string().required(),
+}).required();
+
+export type ArbethTokenFormProps = {
+  onSubmit: (
+    values: ArbethTokenFormValues,
+    formikHelpers: FormikHelpers<ArbethTokenFormValues>
+  ) => void | Promise<void>;
+};
+
+type ArbethTokenFormValues = Yup.Asserts<typeof validationSchema>;
+
+export function ArbethTokenForm({ onSubmit }: ArbethTokenFormProps) {
+  const formik = useFormik<ArbethTokenFormValues>({
+    onSubmit,
+    initialValues: {
+      userKey: '',
+      backupKey: '',
+      tokenContractAddress: '',
+      walletContractAddress: '',
+      walletPassphrase: '',
+      recoveryDestination: '',
+      krsProvider: '',
+      gasLimit: 500000,
+      gasPrice: 30,
+    },
+    validationSchema,
+  });
+
+  const backupKeyHelperText =
+    formik.values.krsProvider === ''
```

Comment:
we don't have krsprovider for arbeth, so this condition can be removed

Reply:
addressed